File: apps/client/src/api/audioSettings.ts
import { get, put } from "./client";

export interface AudioSettings {
  user_id: string;
  input_volume: number;
  output_volume: number;
  input_sensitivity: number;
  voice_activity_enabled: boolean;
  push_to_talk_enabled: boolean;
  push_to_talk_key: string | null;
  noise_suppression_enabled: boolean;
  updated_at: string;
}

export interface UpdateAudioSettingsRequest {
  input_volume?: number;
  output_volume?: number;
  input_sensitivity?: number;
  voice_activity_enabled?: boolean;
  push_to_talk_enabled?: boolean;
  push_to_talk_key?: string | null;
  noise_suppression_enabled?: boolean;
}

export async function getAudioSettings(): Promise<AudioSettings> {
  return get<AudioSettings>("/audio-settings");
}

export async function updateAudioSettings(data: UpdateAudioSettingsRequest): Promise<AudioSettings> {
  return put<AudioSettings>("/audio-settings", data);
}

File: apps/central/src/models/audio_settings.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct AudioSettings {
    pub user_id: Uuid,
    pub input_volume: f32,
    pub output_volume: f32,
    pub input_sensitivity: f32,
    pub voice_activity_enabled: bool,
    pub push_to_talk_enabled: bool,
    pub push_to_talk_key: Option<String>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Deserialize)]
pub struct UpdateAudioSettingsRequest {
    pub input_volume: Option<f32>,
    pub output_volume: Option<f32>,
    pub input_sensitivity: Option<f32>,
    pub voice_activity_enabled: Option<bool>,
    pub push_to_talk_enabled: Option<bool>,
    pub push_to_talk_key: Option<String>,
}
